website/static/js/koHelpers.js

```diff
@@ -205,6 +207,143 @@ ko.bindingHandlers.fadeVisible = {
     }
 };
 
+var tooltip = function(el, valueAccessor) {
+    var params = valueAccessor();
+    $(el).tooltip(params);
+};
+// Run Bootstrap tooltip JS automagically
+// http://getbootstrap.com/javascript/#tooltips
+ko.bindingHandlers.tooltip = {
+    init: tooltip,
+    update: tooltip
+};
+// Attach view model logic to global keypress events
+ko.bindingHandlers.onKeyPress = {
+    init: function(el, valueAccessor) {
+        $(window).keydown(function(e) {
+            var params = valueAccessor();
+            if (e.keyCode === params.keyCode) {
+                params.listener(e);
+            }
+        });
+    }
+};
+
+/* A binding handler to convert lists into formatted lists, e.g.:
+ * [dog] -> dog
+ * [dog, cat] -> dog and cat
+ * [dog, cat, fish] -> dog, cat, and fish
+ *
+ * This handler should not be used for user inputs.
+ *
+ * Example use:
+ * <span data-bind="listing: {data: ['Alpha', 'Beta', 'Gamma'],
+ *                            map: function(item) {return item.charAt(0) + '.';}}"></span>
+ * yields
+ * <span ...>A., B., and G.</span>
+ */
+ko.bindingHandlers.listing = {
+    update: function(element, valueAccessor, allBindings, viewModel, bindingContext) {
+        var value = valueAccessor();
+        var valueUnwrapped = ko.unwrap(value);
+        var map = valueUnwrapped.map || function(item) {return item;};
+        var data = valueUnwrapped.data || [];
+        var keys = [];
+        if (!Array.isArray(data)) {
+            keys = Object.keys(data);
+        }
+        else {
+            keys = data;
+        }
+        var index = 1;
+        var list = ko.utils.arrayMap(keys, function(key) {
```

Comment:
Noted and changed.

Interestingly, the `ko.utils` methods are not documented. They were [intended to be private methods for use internal to the library itself](https://groups.google.com/forum/#!topic/knockoutjs/KpB3SBqg5iI).